### PR TITLE
khalid/ Add new utm fields

### PR DESCRIPTION
**Pull request by Khalidxx**
- Added `utm_ad_id`, `utm_adgroup_id`, and `utm_campaign_id`
- Refactored code

### CODE REVIEW

File: scripts/cookie.js [hunk 1]
```diff
@@ -62,50 +62,40 @@ function toISOFormat(date) {
...
-        ...(utm_campaign && { utm_campaign }),
-        ...(utm_term && { utm_term }),
-        ...(utm_content && { utm_content }),
-      };
+    utm_fields.forEach((field) => {
+      if (searchParams.has(field)) {
+        utm_data[field] = searchParams.get(field).replace(/[^a-zA-Z0-9\s\-\.\_]/gi, '').substring(0, 100);
+      }
+    })
+    console.log(utm_data)
```
Comment (mustofa-binary) [inline scripts/cookie.js:91]: nit: console 😅

File: scripts/cookie.js [hunk 2]
```diff
@@ -62,50 +62,40 @@ function toISOFormat(date) {
...
-      utm_data = {
-        ...(utm_source && { utm_source }),
-        ...(utm_medium && { utm_medium }),
-        ...(utm_campaign && { utm_campaign }),
-        ...(utm_term && { utm_term }),
-        ...(utm_content && { utm_content }),
-      };
+    utm_fields.forEach((field) => {
+      if (searchParams.has(field)) {
+        utm_data[field] = searchParams.get(field).replace(/[^a-zA-Z0-9\s\-\.\_]/gi, '').substring(0, 100);
```
Comment (mustofa-binary) [inline scripts/cookie.js:88]: 😱 isn't this will replace all allowed characters to null

File: scripts/cookie.js [hunk 3]
```diff
@@ -62,50 +62,40 @@ function toISOFormat(date) {
...
-    searchParams.has("utm_campaign")
-  ) {
-    // if url is missing one of required fields, do nothing
-    for (let ctr = 0; ctr < required_fields.length; ctr++) {
-      if (!searchParams.has(required_fields[ctr])) {
-        is_need_change = false;
-      }
-    }
+  // if url is missing one of required fields, do nothing
+  const has_all_params = required_fields.every((field) => searchParams.has(field))
```
Comment (mustofa-binary) [inline scripts/cookie.js:80]: pleae help to check cross browser for `every` 🙏

**Review by mustofa-binary — CHANGES_REQUESTED** [review]

File: scripts/cookie.js (hunk 1 above)
Comment (Khalidxx) [reply scripts/cookie.js:91]: Oh shit, sorry 😅

File: scripts/cookie.js (hunk 2 above)
Comment (Khalidxx) [reply scripts/cookie.js:88]: Oh no, the `^` symbol will select for characters that are not according to the regex. It's actually the same as the binary-static check cause that looked just like the one in the API 😄

File: scripts/cookie.js (hunk 3 above)
Comment (Khalidxx) [reply scripts/cookie.js:80]: Okay checking now

Comment (Khalidxx) [reply scripts/cookie.js:80]: Yess, it works for all modern browsers. Here's the (MDN reference)[https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/every] as well

**Review by mustofa-binary — APPROVED** [review]

**Review by sean-binary — APPROVED** [review]

**Pull request merged**